analytic_engine/src/sst/parquet/meta_data.rs

```diff
@@ -38,68 +45,125 @@ pub enum Error {
 
 define_result!(Error);
 
+const DEFAULT_BLOOM_FILTER_VERSION: u32 = 0;
+
+#[derive(Debug, Clone, PartialEq, Eq, Default)]
+pub struct RowGroupBloomFilter {
+    // The column filter can be None if the column is not indexed.
+    column_filters: Vec<Option<Bloom>>,
+}
+
+impl RowGroupBloomFilter {
+    pub fn with_num_columns(num_columns: usize) -> Self {
+        Self {
+            column_filters: vec![None; num_columns],
+        }
+    }
+
+    pub fn push_column_filter(&mut self, column_filter: Option<Bloom>) {
+        self.column_filters.push(column_filter);
+    }
+
+    /// Accrue the data belonging to one column.
+    ///
+    /// Caller should ensure the `column_idx` is in the range.
+    pub fn accrue_column_data(&mut self, column_idx: usize, data: &[u8]) {
+        if self.column_filters[column_idx].is_none() {
+            self.column_filters[column_idx] = Some(Bloom::default());
+        }
+
+        let column_filter = self.column_filters[column_idx].as_mut().unwrap();
+        column_filter.accrue(Input::Raw(data));
+    }
+
+    /// Return None if the column is not indexed.
+    pub fn contains_column_data(&self, column_idx: usize, data: &[u8]) -> Option<bool> {
+        self.column_filters[column_idx].map(|v| v.contains_input(Input::Raw(data)))
+    }
+}
+
 #[derive(Debug, Clone, PartialEq, Eq, Default)]
 pub struct BloomFilter {
-    // Two level vector means
-    // 1. row group
-    // 2. column
-    filters: Vec<Vec<Bloom>>,
+    // Every filter is a row group filter consists of column filters.
+    //
+    // The row group filter can be None if the row group is not indexed.
```

Comment:
The column of row group can be None?

Reply:
Oops. Forgot to remove this comment after the codes are updated.